webpack.config.js

```diff
@@ -7,6 +8,17 @@ module.exports = {
     filename: 'index.js',
     libraryTarget: 'commonjs2' // THIS IS THE MOST IMPORTANT LINE! :mindblow: I wasted more than 2 days until realize this was the line most important in all this guide.
   },
+  plugins: [
+    new CopyPlugin([
+      // copy the main source file to the `es/index.js.flow` to add
+      // a way to consume flow types for the end users
+      {
+        from: 'src/index.js',
+        to: 'index.js.flow',
```

Comment:
this is neither the best solution (as it produces the extra files in the repo, though it wouldn't affect end users bundle as `.flow` files are only for `flow` type checking), nor the bulletproof solution (e.g. it wouldn't work anyhow if source code will be splitted in the future), but it is the only solution that came into my mind for the moment